plugins/org.eclipse.osee.framework.ui.skynet/src/org/eclipse/osee/framework/ui/skynet/render/NativeRenderer.java

```diff
@@ -90,7 +94,16 @@ public NativeRenderer() {
    public void addMenuCommandDefinitions(ArrayList<MenuCmdDef> commands, Artifact artifact) {
       ImageDescriptor imageDescriptor = null;
       try {
-         imageDescriptor = ImageManager.getProgramImageDescriptor(getAssociatedExtension(artifact));
+         // If artifact has xml extension, extract xml content and create a ProgramImage
+         String extension = getAssociatedExtension(artifact);
+         if (extension.equals("xml")) {
```

Comment:
How about some javadoc explaining this.  Better idea, how about set an attribute once you parse that notates what image to use in future so don't have to parse but once.

Reply:
done